pom.xml

```diff
@@ -137,12 +137,26 @@
             <artifactId>joda-time</artifactId>
             <version>2.10.10</version>
         </dependency>
+        <!-- Testing libraries -->
+		<dependency>
+			<groupId>org.powermock</groupId>
+			<artifactId>powermock-module-junit4</artifactId>
+			<version>2.0.9</version>
```

Comment:
I don't really like having to downgrade Java and mockito because a newer dependency hasn't triggered a release in a few years (despite PRs updating its dependencies).

I don't know what's the alternative though

Reply:
I downgraded to Java 11 to stay consistent with the version used in the pom.xml file.

As for Mockito, I agree... I don't like it either. Unfortunately, the way I found to mock internal classes (without dependency injection) was with Powermock, which is not compatible with newer versions of Mockito.
https://github.com/powermock/powermock/wiki/Mockito#supported-versions

Alternatives to get rid of PowerMock and use the latest version of Mockito:
1. Have an additional constructor on the classes we need to test so we can inject dependencies (like explained in the description)... feels like a crime to me.
2. Have 2 classes for the same Service (`StreamServiceInternal` and `StreamService`). The internal one would work with dependency injection and the public one would simply instantiate the internal one. That way, we could easily test `StreamServiceInternal`:
```java
class StreamServiceInternal {
    public StreamServiceInternal(StreamEnabledSource source, DocumentUploadQueue queue, PlatformClient platformClient) {
        this.source = source;
        this.queue = queue;
        this.platformClient = platformClient;
    }
}
```

```java
class StreamService {
    public StreamService(StreamEnabledSource source) {
        // ...
        this.service = new StreamServiceInternal(source, queue, platformClient);
    }

    public void add(DocumentBuilder document) throws IOException, InterruptedException {
        this.service.add(document);
    }

    public HttpResponse<String> close() throws IOException, InterruptedException, NoOpenStreamException {
        this.service.close();
    }
}
```

Reply:
As ugly as it looks, I think having an internal class where all the logic resides would be the way to go.